pkgs/tools/networking/openssh/common.nix

```diff
@@ -118,19 +116,21 @@ stdenv.mkDerivation rec {
     cat > $DUMMY_PASSWD <<EOF
     $(whoami)::$(id -u):$(id -g)::$HOME:$SHELL
     EOF
-
+    '' + lib.optionalString (!stdenv.hostPlatform.isStatic) ''
```

Comment:
Are you disabling this piece only to avoid reference to (presumably broken) `pkgsStatic.libredirect`? I don't understand how the tests pass without enabling the dummy /etc/passwd

Reply:
Yes. I disable reference to libredirect which do not make sense with static build. I was surprised that build passed without disabling any tests, but it does.

Maybe comment is no longer relevant?

Reply:
I can't tell why, but the code in preCheck is not even run for the static build. If it was, it would fail on the line `cp $HOME/.ssh/environment.base $HOME/.ssh/environment` because that file would not exist. Probably better disable all of preCheck, so not make an impression that any of it is relevant.

Reply:
Oh, `pkgsStatic` is counted as a cross build:
https://github.com/NixOS/nixpkgs/blob/4818f809a92b2ef8d8c002c10a29f1080106ee81/pkgs/stdenv/generic/make-derivation.nix#L103

Reply:
Well, yes. Don't preCheck run on cross builds?

Reply:
No, the checkPhase is skipped entirely.

Reply:
Ok, skipped preCheck entirely.